### PR TITLE
Add form button

### PR DESCRIPTION
**Pull request by chrilehner**
Some questions I have:
- I changed the value property to optional, because it does not make sense for a button element, however the button needs to be an IFormElement to be able to append it to the elements inside the FormBuilder. Is this okay or is there another way of achieving this?
- I made the FormButton class a subclass of EventHandler and now I try to add a click event listener (with `this.paramForm.getElementById(BUTTON_ID).on('click', handler);`), but somehow it isn't called. do I need to do something else inside the FormButton class?

### CODE REVIEW

File: src/form/internal/FormButton.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import {FormElementType, IFormElement, IFormElementDesc, IFormParent} from '../interfaces';
+import * as d3 from 'd3';
+import {EventHandler} from 'phovea_core/src/event';
+
+export interface IButtonElementDesc extends IFormElementDesc {
+  iconClasses: string;
+}
+
+export default class FormButton extends EventHandler implements IFormElementDesc, IFormElement {
```
Comment (sgratzl) [inline src/form/internal/FormButton.ts]: why implementing `IFormElementDesc`?

File: src/form/internal/FormButton.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  /**
+   * Set the visibility of an form element - needed by IFormElement
+   * @param visible
+   */
+  setVisible(visible: boolean) {
+    this.$node.classed('hidden', !visible);
+  }
+
+  build() {
+    this.$button = this.$node.append('button').classed('btn btn-primary', true);
```
Comment (sgratzl) [inline src/form/internal/FormButton.ts]: you aren't registering the handler on the button, do you?

File: src/form/internal/FormButton.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import {FormElementType, IFormElement, IFormElementDesc, IFormParent} from '../interfaces';
+import * as d3 from 'd3';
+import {EventHandler} from 'phovea_core/src/event';
+
+export interface IButtonElementDesc extends IFormElementDesc {
+  iconClasses: string;
```
Comment (sgratzl) [inline src/form/internal/FormButton.ts]: I would add a `onClick` handler directly as an option

File: src/form/internal/FormButton.ts (hunk 2 above)
Comment (chrilehner) [inline src/form/internal/FormButton.ts]: yeah, after I wrote this I tried to do it like
```
this.$button.on('click.propagate', () => {
    this.fire('click', this.$button);
});
```

which I saw in FormInputText

But I like passing the change listener, like you suggested, more

File: src/form/internal/FormButton.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+   */
+  setVisible(visible: boolean) {
+    this.$node.classed('hidden', !visible);
+  }
+
+  get value() {
+    return this.clicked;
+  }
+
+  set value(clicked) {
```
Comment (sgratzl) [inline src/form/internal/FormButton.ts]: typing, please

File: src/form/internal/FormButton.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import {FormElementType, IFormElement, IFormElementDesc, IFormParent} from '../interfaces';
+import * as d3 from 'd3';
+import {EventHandler} from 'phovea_core/src/event';
+
+export interface IButtonElementDesc extends IFormElementDesc {
+  iconClasses: string;
```
Comment (sgratzl) [inline src/form/internal/FormButton.ts]: should be declared optional, since you check if it is set anyhow

File: src/form/internal/FormButton.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  set value(clicked) {
+    this.clicked = clicked;
+  }
+
+  build() {
+    this.$button = this.$node.append('button').classed(this.desc.attributes.clazz, true);
+    this.$button.html(() => this.desc.iconClasses? `<i class="${this.desc.iconClasses}"></i> ${this.desc.label}` : this.desc.label);
+    this.$button.on('click', () => {
+      this.value = true;
+      this.desc.onClick();
```
Comment (sgratzl) [inline src/form/internal/FormButton.ts]: I would add the standard: preventDefault, stopPropagation avoiding side effects

**Pull request merged**